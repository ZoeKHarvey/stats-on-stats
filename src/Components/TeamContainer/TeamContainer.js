import React, { Component } from 'react';
import TeamCard from '../TeamCard/TeamCard'
import { connect } from 'react-redux';
import TeamRoster from '../RosterCard/RosterCard';
// import ice from '../../images/ice4'
import './TeamContainer.scss'



export const TeamContainer = ({teams, roster, getSingleRoster, getSingleTeamSchedule, handlelinkclicks}) => {
  const alph = teams.sort((a,b) => {
   return a.division.id - b.division.id
  })
  console.log('alphabetized--->', alph)
  const singleTeams = alph.map((al) => {
    return <TeamCard key={al.id}
                        {...al}
                        getSingleRoster={getSingleRoster} 
                        getSingleTeamSchedule={getSingleTeamSchedule}
                        handlelinkclicks={handlelinkclicks}
                         />
})

    return(
      <section className="section-teamcontainer">
        
        {singleTeams}
        <h1>Team List</h1>
    
      </section>
    )

}

export const mapStateToProps = (state) => ({
  teams: state.teams
})

export default connect(mapStateToProps)(TeamContainer);