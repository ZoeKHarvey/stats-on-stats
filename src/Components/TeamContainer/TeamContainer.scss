.section-teamcontainer {
  background-image: url('../../images/ice4.jpg')
}