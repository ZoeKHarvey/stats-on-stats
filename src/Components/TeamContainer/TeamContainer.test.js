import React from 'react';
import { TeamContainer, mapStateToProps } from './TeamContainer';
import { shallow } from 'enzyme';

describe('TeamContainer', () => {
  let wrapper;

  let teams = [{
    name: 'Veet',
  }];

  beforeEach(() => {
    wrapper = shallow(
      <TeamContainer
        teams={teams}
        />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mockStateToProps', () => {
  it('should return an object with the teams array', () => {
    const mockState = {
      teams: [{name:'Veet'}, {name: 'Doggan'}]
    };
    const expected = {
      teams: [{name:'Veet'}, {name: 'Doggan'}]
    };
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  })
})

