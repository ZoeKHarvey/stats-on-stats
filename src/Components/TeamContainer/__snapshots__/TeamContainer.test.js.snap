// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamContainer should match the snapshot 1`] = `ShallowWrapper {}`;
